### PR TITLE
fix(replay): post replay views with timestamp = finished_at

### PR DESCRIPTION
**Pull request by aliu39**
Fixes an issue where posting a ReplayViewedEvent modifies the replay's finished_at time (max(timestamp)).
Will also cause POST to 404 if finished_at query returns None (replay doesn't exist)

relates to https://github.com/getsentry/sentry/issues/64924

### CODE REVIEW

File: src/sentry/replays/endpoints/project_replay_viewed_by.py [hunk 1]
```diff
@@ -103,7 +106,32 @@ def post(self, request: Request, project: Project, replay_id: str) -> Response:
...
+        finished_at_ts = None
+        if finished_at_response:
+            finished_at = finished_at_response[0]["finished_at"]
+            finished_at_ts = datetime.fromisoformat(finished_at).timestamp()
+
+        message = viewed_event(
+            project.id,
+            replay_id,
+            request.user.id,
+            finished_at_ts if finished_at_ts else time.time(),
```
Comment (cmanallen) [inline src/sentry/replays/endpoints/project_replay_viewed_by.py:133]: If we can't find the replay let's abort 404 rather than default.

**Review by cmanallen — CHANGES_REQUESTED** [review]

File: src/sentry/replays/endpoints/project_replay_viewed_by.py [hunk 2]
```diff
@@ -103,9 +105,33 @@ def post(self, request: Request, project: Project, replay_id: str) -> Response:
...
+        finished_at_response = execute_query(
+            query=make_full_aggregation_query(
+                fields=["finished_at"],
+                replay_ids=[replay_id],
+                project_ids=[project.id],
+                period_start=filter_params["start"],
+                period_end=filter_params["end"],
+                request_user_id=request.user.id,
+            ),
+            tenant_id={"organization_id": project.organization.id} if project.organization else {},
```
Comment (JoshFerge) [inline src/sentry/replays/endpoints/project_replay_viewed_by.py:119]: i don't think project.organization would ever not exist?

**Review by JoshFerge — APPROVED** [review]

**Review by cmanallen — APPROVED** [review]

File: src/sentry/replays/endpoints/project_replay_viewed_by.py (hunk 2 above)
Comment (aliu3ntry) [reply src/sentry/replays/endpoints/project_replay_viewed_by.py:119]: Not sure.. I copied this pattern from other functions in usecases/query.py, might be good to be safe

Comment (JoshFerge) [reply src/sentry/replays/endpoints/project_replay_viewed_by.py:119]: > might be good to be safe

https://github.com/getsentry/sentry/blob/8666d0c9f60be94988545522e1ff74aa7c0002f5/src/sentry/models/project.py#L229

organization is a required field on project, so i'm not sure this is doing much for us -- would also be curious what's up with usecases/query.py as well.

Comment (aliu3ntry) [reply src/sentry/replays/endpoints/project_replay_viewed_by.py:119]: Ah ok.. thanks for finding a reference. I can follow up in a pr to refactor query code?

Comment (JoshFerge) [reply src/sentry/replays/endpoints/project_replay_viewed_by.py:119]: nbd, if you have a moment, if not no worries

**Pull request merged**